src/main/java/ui/MenuControl.java

```diff
@@ -310,7 +303,15 @@ private MenuItem createRefreshMenuItem() {
         MenuItem refreshMenuItem = new MenuItem("Refresh");
         refreshMenuItem.setOnAction((e) -> {
             logger.info("Menu: View > Refresh");
-            ui.logic.refresh();
+            if (ui.isNotificationPaneShowing()) {
+                // we trigger the notification timeout action first before refreshing
+                Platform.runLater(() -> {
```

Comment:
Is this needed? The menu item will always be clicked/navigated to from the UI thread.